### ISSUE TITLE
Plotnine does not support second axis (sec_axis) like ggplot

### ISSUE THREAD

**Issue by douglasRcummings** [has2k1/plotnine#551]
This is a feature request. It would be awesome if plotnine supported dual y-axes!

**Comment by has2k1** [comment]
This will come. First, we need some refactoring and better layout management to do it well.

**Comment by douglasRcummings** [comment]
curious if this has been added to a roadmap @has2k1 - I can still be patient :)

**Comment by has2k1** [comment]
@douglasRcummings, there is no explicit roadmap yet. But it is on the agenda once it becomes feasible to implement it. That will probably be the next version or the one after.

**Comment by MartinBernstorff** [comment]
Hi `has2k1`! Any news on this currently? :-)

**Comment by DavidKubek** [comment]
I am here to ask the same question. Can we rely on adding this feature or should we (hope not) get used to anothersolution/libraries?

**Comment by has2k1** [comment]
I hope to get this done in a few months. What made it really hard to do is out of the way.

**Comment by MartinBernstorff** [comment]
Thanks for the update on this and your amazing work! It's an incredible one-man-army effort 👍

**Comment by stucash** [comment]
now few months passed...

**Comment by alexlusak** [comment]
Any further updates on this? Similar to other users, converting some company processes to plot natively in Python rather than Excel and could use a secondary axis for some features.

Coming from a plotting intense job in R it's been great to not have to lose all of my built up knowledge, so thanks for the work thus far!

**Comment by PierXuY** [comment]
Any further updates on this?

**Comment by PierXuY** [comment]
A temporary alternative:
```python
%config InlineBackend.figure_format="svg" #jupyter notebook
import pandas as pd
import numpy as np
from plotnine import *
theme_set(theme_classic() + theme(panel_grid_major=element_line(color="#CCCCCC")))

df = pd.DataFrame({
    "col1": np.arange(11),
    "col2": np.arange(11)
})

fig = (
    ggplot(df, aes(x="col1", y="col2", color="col1"))
    + geom_point(size=3)
    + theme(legend_position=(1.2, 0.5))
)

fig_plt = fig.draw()
ax1 = fig_plt.axes[0]
ax2 = ax1.twinx()
ax2.plot([2,7,10],[25,75,120])


# Adjust to a consistent style
ax1.spines["top"].set_visible(False)
ax2.spines["top"].set_visible(False)
ax2.spines["left"].set_visible(False)  

ax1.spines["left"].set_linewidth(1)    
ax2.spines["right"].set_linewidth(1)  

ax1.tick_params(labelsize=10, width=1)
ax2.tick_params(labelsize=10, width=1)

# Show the plot
fig_plt
```

**Comment by TyberiusPrime** [comment]
Hey, 

I know this is in the works, and I'm really looking forward to it.

Any chance the upcoming chances will also allow multiple color/fill scales?

**Comment by has2k1** [comment]
> Any chance the upcoming chances will also allow multiple color/fill scales?

No. And I think that should be an extension or a recipe of sorts.

**Comment by has2k1** [comment]
v0.16.0 will have secondary-axes. And you can already install the [alpha](https://github.com/has2k1/plotnine/issues/1031) version.
